lib/api/core/httpRouter/index.js

```diff
@@ -140,36 +143,45 @@ class Router {
 
     httpRequest.url = httpRequest.url.replace(LeadingSlashRegex, '');
 
-    const routeHandler = this.routes[httpRequest.method].getHandler(httpRequest);
-    routeHandler.getRequest().response.setHeaders(this.defaultHeaders);
+    try {
+      const routeHandler = this.routes[httpRequest.method].getHandler(httpRequest);
 
-    if (routeHandler.handler === null) {
-      return replyWithError(cb, routeHandler.getRequest(), new NotFoundError(`API URL not found: ${routeHandler.url}`));
-    }
+      request = routeHandler.getRequest();
+      request.response.setHeaders(this.defaultHeaders);
 
-    if (httpRequest.content.length <= 0) {
-      return routeHandler.invokeHandler(cb);
-    }
+      if (routeHandler.handler === null) {
+        return replyWithError(cb, request, new NotFoundError(`API URL not found: ${routeHandler.url}`));
+      }
 
-    if (httpRequest.headers['content-type']
-      && !httpRequest.headers['content-type'].startsWith('application/json')) {
-      return replyWithError(cb, routeHandler.getRequest(), new BadRequestError(`Invalid request content-type. Expected "application/json", got: "${httpRequest.headers['content-type']}"`));
-    }
+      if (httpRequest.content.length <= 0) {
+        return routeHandler.invokeHandler(cb);
+      }
+
+      if (httpRequest.headers['content-type']
+        && !httpRequest.headers['content-type'].startsWith('application/json')) {
+        return replyWithError(cb, request, new BadRequestError(`Invalid request content-type. Expected "application/json", got: "${httpRequest.headers['content-type']}"`));
+      }
 
-    {
-      const encoding = CharsetRegex.exec(httpRequest.headers['content-type']);
+      {
+        const encoding = CharsetRegex.exec(httpRequest.headers['content-type']);
 
-      if (encoding !== null && encoding[1].toLowerCase() !== 'utf-8') {
-        return replyWithError(cb, routeHandler.getRequest(), new BadRequestError(`Invalid request charset. Expected "utf-8", got: "${encoding[1].toLowerCase()}"`));
+        if (encoding !== null && encoding[1].toLowerCase() !== 'utf-8') {
+          return replyWithError(cb, request, new BadRequestError(`Invalid request charset. Expected "utf-8", got: "${encoding[1].toLowerCase()}"`));
+        }
       }
-    }
 
-    try {
       routeHandler.addContent(httpRequest.content);
       routeHandler.invokeHandler(cb);
     }
     catch (e) {
-      replyWithError(cb, routeHandler.getRequest(), new BadRequestError('Unable to convert HTTP body to JSON'));
+      if (request === undefined) {
+        request = new Request({requestId: httpRequest.requestId}, {}, 'rest');
+        request.response.setHeaders(this.defaultHeaders);
+      }
+      if (! e instanceof KuzzleError) {
+        e = new InternalError(e);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the caught exception "e". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1226)